src/sync/app.hpp

```diff
@@ -305,26 +315,35 @@ class App {
     /// @param user the user to remove
     /// @param completion_block Will return an error if the user is not found
     void remove_user(std::shared_ptr<SyncUser> user,
-                     std::function<void(Optional<AppError>)> completion_block) const;
+                     std::function<void(util::Optional<AppError>)> completion_block) const;
 
     // Get a provider client for the given class type.
     template <class T>
     T provider_client() {
         return T(this);
     }
 
+    class Internal {
```

Comment:
Is this a common pattern? I'm assuming that this is being done to encapsulate this method exclusively for the friend class `realm::SyncSession` (as opposed to making this a private method which would overexpose App to SyncSession). But this smells a bit. Maybe a SyncSession should simply be constructed with the app route?

Reply:
We poke holes in the private API in a couple of other places like this in Object Store, such as `Realm`.